File: Contracts/Model/Product/UserInteract/Response/OrderResponse.cs
using Contracts.Model.Enterprise;
using Contracts.Model.Security.Response;

namespace Contracts.Model.Product.UserInteract.Response;

public class OrderResponse : BaseResponse
{
	public Guid Id { get; set; } = Guid.NewGuid();
	public string OrderId { get; set; }
	public string Status { get; set; }
	public string UserName { get; set; }
	public ICollection<PurchaseDetailOrderResponse> Details { get; set; }
	public string TransactionId { get; set; }
	public string CurrencyCode { get; set; }
	public string PayerId { get; set; } 
	public double Shipping { get; set; }
	public double Total { get; set; }
	public AddressDto? Address { get; set; }
	public QuoteDetailPreviewResponse? Quote { get; set; }
	public DateTimeOffset? TransactionDate { get; set; }
}

File: ClientApp/src/app/core/interfaces/Product/UserInteract/Response/orderResponse.ts
import {BaseResponse} from "../../../baseResponse";
import {AddressDto} from "@core/interfaces/Product/UserInteract/AddressDto";
import {QuoteDetailPreviewResponse} from "@core/interfaces/Product/UserInteract/Response/QuoteDetialPreviewResponse";
import {PurchaseDetailOrderResponse} from "@core/interfaces/Product/UserInteract/Response/PurchaseDetailOrderResponse";

export interface OrderResponse extends BaseResponse {
  id: string;
  orderId: string;
  status: string;
  userName: string;
  details: PurchaseDetailOrderResponse[];
  transactionId: string;
  currencyCode: string;
  payerId: string;
  shipping: number;
  total: number;
  address: AddressDto;
  quote: QuoteDetailPreviewResponse;
  transactionDate: string;
}
